dev_tools/oneapi/convert_source.sh

```diff
@@ -0,0 +1,429 @@
+replace_regex="${replace_regex};s/blockIdx\.y/item_ct1.get_group(1)/g"
+replace_regex="${replace_regex};s/blockIdx\.z/item_ct1.get_group(0)/g"
+
+replace_regex="${replace_regex};s/blockDim\.x/item_ct1.get_local_range().get(2)/g"
+replace_regex="${replace_regex};s/blockDim\.y/item_ct1.get_local_range().get(1)/g"
+replace_regex="${replace_regex};s/blockDim\.z/item_ct1.get_local_range().get(0)/g"
+
+replace_regex="${replace_regex};s/gridDim\.x/item_ct1.get_group_range(2)/g"
+replace_regex="${replace_regex};s/gridDim\.y/item_ct1.get_group_range(1)/g"
+replace_regex="${replace_regex};s/gridDim\.z/item_ct1.get_group_range(0)/g"
+# Workaround for abs
+replace_regex="${replace_regex};s/sycl::fabs/std::abs/g"
+# Remove unneed warning DPCT1049 - check the block size
+# replace_regex="${replace_regex};/\/\*$/,/DPCT1049/s/^(.*)$/\1-remove/g;/\/\*-remove/,/\*\/$/d"
+replace_regex="${replace_regex};/\/\*$/{N;N;/ *\/\*\n *DPCT1049.*\n *\*\//d}"
+# Recover // template GKO_...
+replace_regex="${replace_regex};s/\/\/ (template GKO.*;)/\1/g"
+# Recover // xxx;//.sync();->xxx.sync();
+replace_regex="${replace_regex};s/;\/\/\.sync\(\)/\.sync\(\);/g"
+replace_regex="${replace_regex};s/\/\/remove\/\///g"
+
+quote="'"
+# echo "replace_regex ${quote}${replace_regex}${quote}"
+sed -i -E "${replace_regex}" "${dpct_file}"
+
+# auto dpct_global_range = grid * block;
+# auto dpct_local_range = block;
+# sycl::nd_range<3>(sycl::range<3>(dpct_global_range.get(2), dpct_global_range.get(1), dpct_global_range.get(0)), sycl::range<3>(dpct_local_range.get(2), dpct_local_range.get(1), dpct_local_range.get(0))),
+
+# check whether containing __dpct_inline__ or __dpct_align__
+need_dpct=$(grep -Eq "__(dpct_align|dpct_inline)__" ${dpct_file})
+if grep -Eq "__(dpct_align|dpct_inline)__" ${dpct_file}; then
+    # Found
+    sed -i '1 i#include "dpcpp/base/dpct.hpp"' ${dpct_file}
+fi
+
+# extract device_code
+if [ "${EXTRACT_KERNEL}" = "true" ]; then
+    IFS=';' read -ra individual_deivce <<< "${DEVICE_FILE}"
+    for variable in "${individual_deivce[@]}"; do
+        device_regex=$(convert_regex_allowed "${variable}")
+        # echo "device_regex ${device_regex}"
+        dpct_device_path=$(echo "${variable}" | sed 's/common/dpcpp_code/g')
+        dpct_device_file=$(echo "${dpct_device_path}" | sed 's/\//@/g')
+        dpct_device_file="output/${dpct_device_file}"
+        # echo "dpct_device_path ${dpct_device_path}"
+        # echo "dpct_device_file ${dpct_device_file}"
+        cat ${dpct_file} | sed -n "/${device_regex} - start/,/${device_regex} - end/p" | sed "1d;\$d" > ${dpct_device_file}
+        sed -i "/${device_regex} - start/,/${device_regex} - end/d;s~// *#include \"${device_regex}\"~#include \"${dpct_device_path}\"~g" ${dpct_file}
+        dpct_dir=$(dirname "${dpct_device_path}")
+        mkdir -p "${ROOT_DIR}/${dpct_dir}"
+        cp "${dpct_device_file}" "${ROOT_DIR}/${dpct_device_path}"
+    done
+fi
```

Comment:
I fail to see for now, what is this for?

Reply:
it is optional and off.
it will move the common code to common/dpcpp_code if enabling it.